crates/rpc/admin/mod.rs

```diff
@@ -1,21 +1,48 @@
-use serde_json::{json, Value};
+use ethereum_rust_core::types::ChainConfig;
+use ethereum_rust_net::types::Node;
+use serde::Serialize;
+use serde_json::Value;
+use std::collections::HashMap;
 
 use crate::utils::RpcErr;
 
-pub fn node_info() -> Result<Value, RpcErr> {
-    Ok(json!({
-        "enode": "enode://pubkey@ip:port",
-        "id": "pubkey",
-        "name": "node",
-        "ports": {
-            "discovery": 1234,
-            "listener": 1234,
-        },
-        "protocols": {
-            "eth": {
-                "network": 1234,
-                "version": 1234,
-            },
+#[derive(Serialize, Debug)]
+struct NodeInfo {
+    enode: String,
+    id: String,
+    ip: String,
+    name: String,
+    ports: Ports,
+    protocols: HashMap<String, Protocol>,
+}
+
+#[derive(Serialize, Debug)]
+struct Ports {
+    discovery: u16,
+    listener: u16,
+}
+
+#[derive(Serialize, Debug)]
+#[serde(untagged)]
+enum Protocol {
+    Eth(ChainConfig),
+}
+
+pub fn node_info(chain_config: ChainConfig, local_node: Node) -> Result<Value, RpcErr> {
```

Comment:
instead of passing chain config as a param/context, you should be able to get it from storage

Reply:
done!